src/api/app/views/webui/project/_tabs.html.haml

```diff
@@ -33,9 +33,15 @@
       %li.nav-item
         = tab_link('Pulse', project_pulse_path(project))
       - active = controller_path.starts_with?('webui/staging')
-      - if (!project.staging_project? && policy(Staging::Workflow.new(project: project)).create?) || active
+      - if project.staging_project?
+        %li.nav-item
+          = tab_link('Staging', staging_workflow_path(project.staging_workflow), active)
+      - elsif policy(Staging::Workflow.new(project: project)).create?
         %li.nav-item
           = tab_link('Staging', new_staging_workflow_path(project: project), active)
+      - elsif (staging = Staging::Workflow.find_by(project_id: project.id))
```

Comment:
Using `project.staging` doesn't work here because the association returns a `Staging::Workflow` even when there isn't one for the project. It returns an empty `Staging::Workflow`, except for the `project_id` field which contains the value of `project.id`. I don't understand why it does this to be honest.

Reply:
Hmmm :thinking:

Reply:
I can't reproduce:

```irb(main):002:0> Project.find_by(name: 'home:coolo').staging
=> nil

Reply:
This happens after `policy(Staging::Workflow.new(project: project)).create?` is executed. To reproduce it in the view:

1. Add `binding.pry` above `- if project.staging_project?` (line 36 in the changed file)
2. Go to the project#show page of a project with a staging workflow.
3. Execute `project.staging` in `pry`, see the `Staging::Workflow` as expected:
```
#<Staging::Workflow:0x00007f2e3a6a8208 id: 3, project_id: 28, created_at: Wed, 25 Sep 2019 11:19:06 UTC +00:00, updated_at: Wed, 25 Sep 2019 11:19:06 UTC +00:00, managers_group_id: 4>
```
4. Execute `policy(Staging::Workflow.new(project: project)).create?` in `pry`
5. Execute again `project.staging` in `pry`, see the `Staging::Workflow`:
```
#<Staging::Workflow:0x00007f2e3b9a1d50 id: nil, project_id: 28, created_at: nil, updated_at: nil, managers_group_id: nil>
```

Kudos to @DavidKang for pointing me to the cause (the policy). I still don't see why it does this though.

Reply:
I see - `Staging::Workflow::new(project:project)` is binding the new object to the `belongs_to` association. But then make it simple and use project.staging_workflow_id instead of running a N+1 query (where N is fortunately 1 :)

Reply:
`project.staging_workflow_id` is for the `staging_projects` association of a `Staging::Workflow`, so I cannot use it to check if a project has a staging workflow.

Reply:
right. How about `project.staging.id` then? :)

Reply:
But even that will cause an extra query if you don't prefetch

Reply:
I ended up going with `persisted?`.

Reply:
Can you file an issue for the policy please?